src/Facebook/InstantArticles/Transformer/Rules/ConfigurationSelectorRule.php

```diff
@@ -123,11 +123,14 @@ public function matchesNode($node)
 
         $results = $domXPath->query($xpath);
 
-        foreach ($results as $result) {
-            if ($result === $node) {
-                return true;
+        if ( is_array( $results ) ) {
```

Comment:
@philipjohn This won't work because the method DOMXPath::query doesn't return an array. 
It returns an DOMNodeList (http://php.net/manual/en/class.domnodelist.php) which is just an Traversable with iterator.